src/libraries/System.Private.Xml/src/System/Xml/Core/XmlReader.cs

```diff
@@ -1714,13 +1714,8 @@ internal static XmlReader CreateSqlReader(Stream input, XmlReaderSettings? setti
             // allocate byte buffer
             byte[] bytes = new byte[CalcBufferSize(input)];
 
-            int byteCount = 0;
-            int read;
-            do
-            {
-                read = input.Read(bytes, byteCount, bytes.Length - byteCount);
-                byteCount += read;
-            } while (read > 0 && byteCount < 2);
+            int bytesToRead = Math.Min(bytes.Length, 2);
+            int byteCount = input.ReadAtLeast(bytes, bytesToRead, throwOnEndOfStream: false);
```

Comment:
the previous logic was hard to understand: read at least 2 bytes? now it looks much better 👍